pkg/controller/clusteringress/clusteringress_controller.go

```diff
@@ -103,6 +103,10 @@ func (r *ReconcileClusterIngress) Reconcile(request reconcile.Request) (reconcil
 
 	// Don't modify the informer's copy
 	ci := original.DeepCopy()
+	if len(ci.GetFinalizers()) == 0 {
+		ci.SetFinalizers([]string{"ingress"})
+		return reconcile.Result{}, r.client.Update(context.TODO(), ci)
```

Comment:
Is it necessary to abort reconcilation? Maybe we only need to do that if the Update call fails?